### PR TITLE
Fixed Kotlin generic functions and extension functions coloring

### PR DESCRIPTION
**Pull request by rongi**
Kotlin rendering issue #736
https://github.com/jneen/rouge/issues/736

### CODE REVIEW

File: lib/rouge/lexers/kotlin.rb [hunk 1]
```diff
@@ -56,6 +56,18 @@ class Kotlin < RegexLexer
           groups Keyword::Declaration, Text
           push :property
         end
+        rule %r'(fun)(\s+)(<)(#{name})(>)(\s+)(#{name})(\.)' do
```
Comment (gfx) [inline lib/rouge/lexers/kotlin.rb]: It's too simple.

```kotlin
fun <T, U> f() {}
fun <T>f() { }
```

Don't use long regexp to parse nestable syntax parts.

Comment (rongi) [reply lib/rouge/lexers/kotlin.rb]: @gfx Any suggestions how to make it right? It was pretty challenging to get it even this far. May be you can point me to an example where something like this was implemented?

**Comment by rongi** [comment]
Aight, looks like it's fixed now

**Comment by gfx** [comment]
Regexp can't handle nested syntax.

I've fixed this issue in a simpler way: https://github.com/jneen/rouge/pull/782

**Pull request closed without merging**